include/tscore/OneWriterMultiReader.h

```diff
@@ -0,0 +1,225 @@
+/** @file
+
```

Comment:
Is there a reason this PR rolls its own reader/writer lock rather than using one of the standard implementations?

Reply:
Ah, I didn't carefully read your initial comment.  What is the performance hit of using the std implementation of a reader_writer lock?

Reply:
In the scenario I tested, I saw more than double the requests per second using my shared mutex implementation.

But, in truth, if we go with the second option (where each Debug() call has an instance of DbgCtl, even if it has to create it itself), it will be moot, the lock will almost never need to be locked.

Aside from better performance, my shared mutex has logic to prevent writer starvation.  But it's not recursive.

It was actually a mistake that I implemented my own shared mutex, I didn't realize pthreads (and C++17) had one.  But looking like a lucky accident.

Reply:
One thing I couldn't understand in my performance testing is, no matter how I tweaked the parameters of h2load, I couldn't get under 65% idle on atslab00.  All the requests were the same, the (cleartext) GET of a single object in cache, so maybe ATS was disk I/O bound?

Reply:
An important point to keep in mind is that, in my testing, I used a tag regex that didn't match any tag.  So the 10% drop in throughput did not include the overhead of any actual output.  Running in production with a regex like "http|dns" would result in a much bigger drop in throughput than 10%.  Debugging could only be turned on in production with a regex that enabled a small amount of debug output.